backend/users-service/app/src/index.ts

```diff
@@ -1,7 +1,71 @@
-import { Elysia } from "elysia";
+import { Elysia, t } from 'elysia'
+import { swagger } from '@elysiajs/swagger'
+import { cors } from '@elysiajs/cors'
+import { helmet } from 'elysia-helmet'
+import jwt from '@elysiajs/jwt'
+import { PrismaClient } from '@prisma/client'
+import { logger } from '@bogeychan/elysia-logger'
+import { Role } from './constants'
 
-const app = new Elysia().get("/", () => "Users service").listen(80);
+const db = new PrismaClient()
+
+const app = new Elysia({ prefix: '/users' })
```

Comment:
i'm not familiar with Elysia but does this mean the routes are located under `/users` or `/api/users`? because we would need the latter

Reply:
> i'm not familiar with Elysia but does this mean the routes are located under `/users` or `/api/users`? because we would need the latter

Currently its `/users`. Ok will change to `/api/users`

Reply:
@raynerljm actually is the API gateway's proxy base URL to users service `/api/users`? If so I think we can just remove the `/users` prefix on my side completely

Reply:
@sheimoria discussed this with rayner for my service. Actually /api/users on the API gateway's proxy base URL will reroute to /api/users on your service's URL. so you would need to change this prefix from /users to /api/users.

But following best practices would be good if you can make a tree structure with /api as the root url then /users is one of the subroutes of the /api route. I'm not sure how this is done in elysia though

Reply:
Refactored file structure in 89f565cb95ba774a0b66510f252952527116cfae